e2e/preview/pipelines/test_gradient_rag_pipelines.py

```diff
@@ -0,0 +1,92 @@
+import os
+import json
+import pytest
+
+from haystack.preview import Pipeline, Document
+from haystack.preview.components.embedders.gradient_document_embedder import GradientDocumentEmbedder
+from haystack.preview.components.embedders.gradient_text_embedder import GradientTextEmbedder
+from haystack.preview.document_stores import InMemoryDocumentStore
+from haystack.preview.components.writers import DocumentWriter
+from haystack.preview.components.retrievers import InMemoryEmbeddingRetriever
+from haystack.preview.components.generators.gradient.base import GradientGenerator
+from haystack.preview.components.builders.answer_builder import AnswerBuilder
+from haystack.preview.components.builders.prompt_builder import PromptBuilder
+
+
+@pytest.mark.skipif(
+    not os.environ.get("GRADIENT_ACCESS_TOKEN", None),
```

Comment:
`GRADIENT_WORKSPACE_ID` also needs to be set or otherwise we need to skip the test.

Reply:
done!